openldap/2.6/debian-11/rootfs/opt/bitnami/scripts/libopenldap.sh

```diff
@@ -0,0 +1,1045 @@
+    fi
+}
+
+########################
+# Stop OpenLDAP server
+# Arguments:
+#   $1 - max retries. Default: 12
+#   $2 - sleep between retries (in seconds). Default: 1
+# Returns:
+#   None
+#########################
+ldap_stop() {
+    local -r retries="${1:-12}"
+    local -r sleep_time="${2:-1}"
+
+    are_db_files_locked() {
+        local return_value=0
+        read -r -a db_files <<< "$(debug_execute find "$LDAP_DATA_DIR" -type f -print0 | xargs -0)"
+        for f in "${db_files[@]}"; do
+            result=$(fuser "$f" 2>&1)
+            pid=$(echo $result | cut -d ':' -f 2-)
+            if [ -n "$pid" ]; then
+                if is_boolean_yes "${SYMAS_DEBUG:-false}" ; then
+                    process=$(ps -p $"pid" -o comm=)
+                    warn "ldap_stop waiting on ${process} to close ${f} file"
+                fi
+                return_value=1
+            fi
+        done
+        return "$return_value"
+    }
+
+    is_ldap_not_running && return
+
+    stop_service_using_pid "$LDAP_PID_FILE"
+    if ! retry_while are_db_files_locked "$retries" "$sleep_time"; then
+        error "OpenLDAP failed to stop"
+        return 1
+    fi
+}
+
+########################
+# Create slapd.ldif
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_create_slapd_file() {
+    info "Creating slapd.ldif"
+    cat > "${LDAP_SHARE_DIR}/slapd.ldif" << EOF
+#
+# See slapd-config(5) for details on configuration options.
+# This file should NOT be world readable.
+#
+
+dn: cn=config
+objectClass: olcGlobal
+cn: config
+olcArgsFile: ${LDAP_BASE_DIR}/var/run/slapd.args
+olcPidFile: ${LDAP_BASE_DIR}/var/run/slapd.pid
+
+#
+# Load dynamic backend modules commonly compiled in and available by default:
+#
+dn: cn=module,cn=config
+objectClass: olcModuleList
+cn: module
+olcModulePath: ${LDAP_BASE_DIR}/lib/openldap
+olcModuleLoad: back_mdb.la
+#olcModuleload: back_ldap.la
+#olcModuleload: back_passwd.la
+
+#
+# Schema settings
+#
+
+dn: cn=schema,cn=config
+objectClass: olcSchemaConfig
+cn: schema
+
+include: file://${LDAP_BASE_DIR}/etc/schema/core.ldif
+
+#
+# Frontend settings
+#
+
+dn: olcDatabase=frontend,cn=config
+objectClass: olcDatabaseConfig
+objectClass: olcFrontendConfig
+olcDatabase: frontend
+
+#
+# Configuration database
+#
+
+dn: olcDatabase=config,cn=config
+objectClass: olcDatabaseConfig
+olcDatabase: config
+olcAccess: to * by dn.base="gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth" manage by * none
+
+#
+# Server status monitoring
+#
+
+dn: olcDatabase=monitor,cn=config
+objectClass: olcDatabaseConfig
+olcDatabase: monitor
+olcAccess: to * by dn.base="gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth" read by dn.base="cn=Manager,dc=my-domain,dc=com" read by * none
+
+#
+# Backend database definitions
+#
+dn: olcDatabase=mdb,cn=config
+objectClass: olcDatabaseConfig
+objectClass: olcMdbConfig
+olcDatabase: mdb
+olcDbMaxSize: 1073741824
+olcSuffix: dc=my-domain,dc=com
```

Comment:
Doesn't sound right to put a random suffix in only to change it later.

Reply:
replaced with LDAP_ROOT